### PR TITLE
Convert m files

### PR DESCRIPTION
**Pull request by vdusart**
Moving files starting with letter "M" to TS

# Related Issue

Conversion to TS according to #6392

### CODE REVIEW

File: src/components/MatomoOptOut.tsx [hunk 1]
```diff
@@ -24,29 +24,31 @@ const Container = styled.div`
...
       setIsOptedOut(true)
     } else {
-      localStorage.setItem(MATOMO_LS_KEY, false)
+      localStorage.setItem(MATOMO_LS_KEY, "false")
     }
     setLoading(false)
   }, [])
 
-  const handleCheckbox = ({ target: { checked } }) => {
+  const handleCheckbox = ({ target: { checked } }): void => {
```
Comment (pettinarip) [inline src/components/MatomoOptOut.tsx:46]:
```suggestion
  const handleCheckbox = ({ target: { checked } }: React.ChangeEvent<HTMLInputElement>): void => {
```

File: src/components/Modal.tsx [hunk 2]
```diff
@@ -66,18 +66,33 @@ const ModalCloseIcon = styled(Icon)`
...
     transition={{ duration: 0.2 }}
   />
 )
 
-const Modal = ({ children, className, isOpen, setIsOpen }) => {
-  const ref = useRef()
+export interface IProps {
+  className?: string
+  isOpen: boolean
+  setIsOpen: (boolean: boolean) => void
```
Comment (pettinarip) [inline src/components/Modal.tsx:84]:
```suggestion
  setIsOpen: (isOpen: boolean) => void
```

**Review by pettinarip — APPROVED** [review]
Left two small comments, the rest looks great! gj @vdusart

**Review by pettinarip — CHANGES_REQUESTED** [review]
Missclicked the approve button

**Comment by vdusart** [comment]
@pettinarip I just fixed your comments and add small changes to files that I had made.

**Review by pettinarip — APPROVED** [review]
Cool @vdusart thanks for that!

**Pull request merged**